inc/html.class.php

```diff
@@ -1662,7 +1662,35 @@ static function footer($keepDB = false) {
       $tpl_vars['js_files'][] = 'js/misc.js';
 
       // TODO Add Ajax notifications script block
-      // TODO Add plugins scripts
+      if (isset($PLUGIN_HOOKS['add_javascript']) && count($PLUGIN_HOOKS['add_javascript'])) {
+         foreach ($PLUGIN_HOOKS["add_javascript"] as $plugin => $files) {
+            $plugin_root_dir = Plugin::getPhpDir($plugin, true);
+            $plugin_web_dir  = Plugin::getWebDir($plugin, false);
+            if (!Plugin::isPluginActive($plugin)) {
+               continue;
+            }
+            $version = Plugin::getInfo($plugin, 'version');
```

Comment:
Does not seems to be used.

```suggestion
```

Reply:
> Modules handling is a bit complex. You had to modify many places to handle both strings and arrays containing `file` and optionnal `type`.
> Maybe adding a new hook (`add_javascript_module`) and always passing a `file + optionnal type` array to view should be easier to maintain.

With the new hook, is specifying the type really needed?

Reply:
Would it be better to keep a single hook in case there are other parameters we want to add later? Right now I can think of at least two other parameters that could be nice:
- "context": Load a script from the core or another plugin (assuming the plugin is enabled). Maybe handle de-duplicating requested scripts too.
I think loading core scripts from a plugin can be done differently now from a recent change, but I cannot find that PR now...
- "location" - Specify if the script should be loaded in the head or footer.

Reply:
The rmain equirement here is to be able to load modules, so maybe we should just add a new `add_javascript_module` that takes only a file path, and pass a `js_modules` var to the view.
IMHO, other improvement should only be done in a global JS code improvement of GLPI, which may require specifications to be defined in advance.